crates/hir-ty/src/infer/expr.rs

```diff
@@ -815,7 +815,7 @@ impl<'a> InferenceContext<'a> {
             Expr::Array(array) => self.infer_expr_array(array, expected),
             Expr::Literal(lit) => match lit {
                 Literal::Bool(..) => self.result.standard_types.bool_.clone(),
-                Literal::String(..) => {
+                Literal::String(..) | Literal::CString(..) => {
                     TyKind::Ref(Mutability::Not, static_lifetime(), TyKind::Str.intern(Interner))
                         .intern(Interner)
```

Comment:
The type of c string literals should be `&CStr`, for which [a new lang item `CStr`][lang-item] has been introduced. Also, some hir-ty tests would be nice!

[lang-item]: https://github.com/rust-lang/rust/blob/77c836e1ae582661924d3b6ec4d57a2de120f59f/library/core/src/ffi/c_str.rs#L84

Reply:
Ah, right. For some reason I completely forgot about that 😅